### ISSUE TITLE
DL3060 unwarranted in all but the final stage of multi-stage Dockerfile builds

### ISSUE THREAD

**Issue by robdesbois** [hadolint/hadolint#619]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

I expected DL3060 warnings only for the final stage of multi-stage builds, as having larger earlier stages only affects the build cache, not the final image size.

### Actual behavior

DL3060 warning is flagged for non-final stages that run `yarn install` not followed by `yarn cache clear`.

### Steps to reproduce the behavior

Run `hadolint` on the `Dockerfile` below:

```bash
➜  hadolint --version
Haskell Dockerfile Linter 2.3.0-no-git
```

**Dockerfile** (if relevant)

```Dockerfile
## first stage
FROM node:lts-alpine as builder
WORKDIR /usr

COPY package.json tsconfig.json yarn.lock ./
RUN yarn install --frozen-lockfile

COPY src ./src/
RUN yarn build

## second stage
FROM node:lts-alpine as main
WORKDIR /

COPY /usr/package.json .
COPY --from=builder /usr/node_modules ./node_modules/
COPY --from=builder /usr/dist ./dist/

USER node
ENTRYPOINT ["node", "./dist/index.js"]
EXPOSE 9000
```

**Additional environment details (OS, stack version, etc.)**
Ubuntu 20.04 kernel 5.4.0-72-generic
